packages/query-core/src/queryObserver.ts

```diff
@@ -272,6 +267,14 @@ export class QueryObserver<
       get: (target, key) => {
         this.trackProp(key as keyof QueryObserverResult)
         onPropTracked?.(key as keyof QueryObserverResult)
+        if (key === 'promise' && !this.options.experimental_prefetchInRender) {
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          ;(this.#currentThenable as PendingThenable<TData>).reject?.(
+            new Error(
+              'experimental_prefetchInRender feature flag is not enabled',
+            ),
+          )
```

Comment:
@KATT the type assertion + optional chaining is necessary because we might not have a `PendingThenable` here and the generic `Thenable` doesn’t have a `reject` method :/

I don’t fully understand why that is but I’m still shipping this fix 😂